File: 0463-island-perimeter/0463-island-perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int n = grid.length;
        int m = grid[0].length;
        int f = 0;
        for (int i = 0;i<n;i++)
        {
            for (int j = 0;j<m;j++)
            {
                if (grid[i][j] == 0)
                {
                    
                }
                else
                {
                    if (j == 0)
                    {
                        f++;
                    }
                    if (i == 0)
                    {
                        f++;
                    }
                    if (j == m-1)
                    {
                        f++;
                    }
                    if (i == n-1)
                    {
                        f++;
                    }
                    if (j > 0)
                    {
                        if (grid[i][j-1] == 0)
                        {
                            f++;
                        }
                    }
                    if (j < m-1)
                    {
                        if (grid[i][j+1] == 0)
                        {
                            f++;
                        }
                    }
                    if (i > 0)
                    {
                        if (grid[i-1][j] == 0)
                        {
                            f++;
                        }
                    }
                    if (i < n-1)
                    {
                        if (grid[i+1][j] == 0)
                        {
                            f++;
                        }
                    }
                }
            }
        }
        return f;
    }
}

File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        ROW = len(grid)
        COL = len(grid[0])
        arr = [[0, -1], [-1, 0], [0, 1], [1, 0]]
        res = 0
        for i in range(ROW):
            for j in range(COL):
                if grid[i][j] == 1:
                    for m, n in arr:
                        im = i + m
                        jn = j + n
                        if im < 0 or im >= ROW or jn < 0 or jn >= COL:
                            res += 1
                        elif grid[im][jn] == 0:
                            res += 1
        return res